src/utils/hooks/useUpdateReviewResponseNEW.ts

```diff
@@ -0,0 +1,57 @@
+import {
+  ReviewResponse,
+  ReviewResponseDecision,
+  ReviewResponseRecommendedApplicantVisibility,
+  useUpdateReviewResponseMutation,
+} from '../generated/graphql'
+
+// below lines are used to get return type of the function that is returned by useUpdateReviewResponseMutation
+type UseUpdateReviewMutationReturnType = ReturnType<typeof useUpdateReviewResponseMutation>
+type promiseReturnType = ReturnType<UseUpdateReviewMutationReturnType[0]>
+
+// hook used to update review response, comment and decision, use as per type definition belo
+// also computes and updates recommendedApplicantVisibility
+type useUpdateReviewResponse = (
+  reviewResponseId: number
+) => (reviewResponse: ReviewResponse) => promiseReturnType
+
+const useUpdateReviewResponse: useUpdateReviewResponse = (reviewResponseId) => {
+  const [updateReviewResponse] = useUpdateReviewResponseMutation()
+
+  return async (reviewResponse) =>
+    updateReviewResponse({
+      variables: {
+        id: reviewResponseId,
+        comment: reviewResponse.comment,
+        decision: reviewResponse.decision,
+        recommendedApplicantVisibility: computeVisibility(reviewResponse),
+      },
+    })
+}
+
+// is_visible_to_applicant is set by an action on back, it uses recomended_application_visibility of the
+// last level review to determine what will be visible to application in LOQ
+// we can add a checkbox in decision modal to determine this (thus logic is placed here)
+const computeVisibility = (reviewResponse: ReviewResponse | undefined) => {
+  // would rather computer this from actual review level, but dont' want to pass too far as props or rely on context
+  if (!reviewResponse)
+    return ReviewResponseRecommendedApplicantVisibility.OriginalResponseNotVisibleToApplicant
+
+  const isLevelOne = reviewResponse.originalReviewResponseId === reviewResponse.id
+
+  if (isLevelOne) {
+    if (reviewResponse.decision === ReviewResponseDecision.Decline)
+      return ReviewResponseRecommendedApplicantVisibility.OriginalResponseVisibleToApplicant
+
+    return ReviewResponseRecommendedApplicantVisibility.OriginalResponseNotVisibleToApplicant
+  }
```

Comment:
What about if there is only one level? The responses should become visible to applicant in that case, shouldn't they?

Reply:
Yeah that isLevelOne clause should deal with it

Reply:
Okay, sure, I think I was confused for the same reason as the other comment -- I thought comments for Approved responses should also be visible to applicant.